File: frontend/src/api/quota.ts
// Matches app/schemas/quota.py exactly (issue #20).
import { apiFetch } from "./client";

export interface QuotaProgress {
  count: number;
  target: number;
  remaining: number;
}

export interface QuotaSummary {
  cold_outreach_sent: QuotaProgress;
  warm_intro_requests_sent: QuotaProgress;
  cold_applications_submitted: QuotaProgress;
  referral_asks_made: QuotaProgress;
}

export function getQuotaProgress(): Promise<QuotaSummary> {
  return apiFetch<QuotaSummary>("/quotas");
}

File: backend/app/schemas/quota.py
"""Pydantic schema for GET /api/quotas (issue #20)."""

from pydantic import BaseModel


class QuotaProgress(BaseModel):
    count: int
    target: int
    remaining: int


class QuotaSummary(BaseModel):
    cold_outreach_sent: QuotaProgress
    warm_intro_requests_sent: QuotaProgress
    cold_applications_submitted: QuotaProgress
    referral_asks_made: QuotaProgress
